doc/adr/0011-tsp-analysis-api.md

```diff
@@ -0,0 +1,268 @@
+# 11. Tsp analysis api
+
+Date: 2023-06-20
+
+## Status
+
+Proposed
+
+## Context
+
+The trace viewer currently is able to visualize trace data provided by a trace server over the trace server protocol (TSP). The Trace Compass server has some built-in analysis view for that. However, it is not possible to side-load analysis and visualization descriptions over the TSP so that end-user can provide some custom views. The Trace Compass supports loading of data-driven analysis and views, e.g. XML driven views or in-and-out anlysis of the Trace Compass incubator project. In the Eclispe-based Trace Compass application, there exists UI primitive to load e.g. XML files or configure custom analysis. While the Trace Compass server has the capablility to understand these defintions, there is no way to side-load this definition over the TSP. This ADR will propose an analysis service within the TSP to facilitate these custom analysis.
+
+### Global analysis service
+
+An analysis service for managing global analyses will be introduced. Global in this context means that the analysis definitions will be handled on the application level. 
+
+    GET /analysis/
+        returns a list of analysis type descriptors: typeId, name, description, expected input parameter list ("path" for file path)
+    GET /analysis/types
+        returns a map typeId -> list of analysis descriptor of existing analyses on server: analysisId, name, description, parameters
+    GET /analysis/types/{typeId}
+        returns a list of analysis descriptors for given typeId
+    POST /analysis/types/{typeId}
+        Upload an analysis definition for given typeId
+        Returns a new analysis descriptor with unique analysisID
+    PUT /analysis/types/{typeId}/{analysisId}
+        Update an analysis
+    GET /analysis/types/{typeId}/{analysisId}
+        Returns the analysis descriptor for given analysisId
+    DELETE /analysis/types/{typeId}/{analysisId}
+        Delete an analysis
+
+#### Analysis type descriptor
+
+The analysis type descriptor describes the type of external analysis. Different types have different syntax and hence back-end implementation. This descriptor is used to distinquish the different external analysis. The trace server implementation will provide the list of type descriptors on client requests.
+```javascript
+AnalysisTypeDescriptor {
+    name: string,
+    description: string,
+    id: string,
+    scope: string,
+    inputParameter: string[]
+}
+```
+
+Where:
+
+- `name`: Name of the analysis type. Can be shown to the end-user.
+- `description`: The description of the analysis type. Can be shown to the end-user.
+- `id`: Unique id of the analysis type. Used in the application to distinquish analysis types
+- `scope:` `experiment` for analysis per experiment or `global` for all experiments
```

Comment:
Trying to understand how the scope would be used. From the sequences below, it seems that the experiment analysis descriptors are accessed using different endpoints with experiment/{expUUID} path. But these sequences do not seem to use any steps to get the AnalysisTypeDescriptor.

Reply:
The AnalysisTypeDescriptor still go through the global end-point, where you will be able to see the types and create analysis descriptors. Then the experiment endpoint is to basically assign an analysis descriptor to an experiment (see post). With this you could assign a given analysis to multiple selected experiments, I think I have to update "Sequence: Create analysis instance for an experiment", because it's not really reflecting that.